### PR TITLE
Add compliance test for UBOs sized over 65536

### PR DESCRIPTION
**Pull request by JasperNV**
https://bugs.chromium.org/p/angleproject/issues/detail?id=3388

### CODE REVIEW

File: sdk/tests/conformance2/uniforms/large-uniform-buffers.html [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    var uboData = new Float32Array(0x20000);
+
+    var offs0 = 0x00000;
+    // Red
+    uboData[offs0 + 0] = 1;
+    uboData[offs0 + 1] = 0;
+    uboData[offs0 + 2] = 0;
+    uboData[offs0 + 3] = 1;
+
+    var offs1 = 0x10000;
```
Comment (kenrussell) [inline sdk/tests/conformance2/uniforms/large-uniform-buffers.html]: offs1 has to be computed as a multiple of gl.getParameter(UNIFORM_BUFFER_OFFSET_ALIGNMENT), or the test might be incorrect on some platforms.

File: sdk/tests/conformance2/uniforms/large-uniform-buffers.html [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+    var uboDataSize = gl.getActiveUniformBlockParameter(
+        program, uboIndex, gl.UNIFORM_BLOCK_DATA_SIZE);
+    if (uboDataSize == 0) {
+        testFailed("uniform block data size invalid");
+        return;
+    }
+
+    var uboBuf = gl.createBuffer();
+    var uboData = new Float32Array(0x20000);
```
Comment (kenrussell) [inline sdk/tests/conformance2/uniforms/large-uniform-buffers.html]: There aren't any queries of gl.getParameter(MAX_UNIFORM_BLOCK_SIZE) in this test. Should there be? I realize that the program is only reading 4 floats out of the UBO. We have seen this 65K UBO size limit on other platforms too (macOS) and it's accounted for in the UBO rendering path in this project for example:
https://github.com/kenrussell/webgl-animometer

File: sdk/tests/conformance2/uniforms/large-uniform-buffers.html (hunk 1 above)
Comment (JasperNV) [reply sdk/tests/conformance2/uniforms/large-uniform-buffers.html]: The maximum alignment ever seen in the wild is 256 (by NVIDIA GPUs): http://opengl.gpuinfo.org/displaycapability.php?name=GL_UNIFORM_BUFFER_OFFSET_ALIGNMENT , and there is also GLES spec language capping it to 256. The `65536` size isn't an OpenGL limit, it's a Direct11 cbuffer limit, so there's no easy way I can compute this offset from the details given to me from GLES alone.

File: sdk/tests/conformance2/uniforms/large-uniform-buffers.html (hunk 2 above)
Comment (JasperNV) [reply sdk/tests/conformance2/uniforms/large-uniform-buffers.html]: Based on my reading of the GLES spec, GL_MAX_UNIFORM_BLOCK_SIZE describes the maximum size of any individual GL block used by the shader. It does *not* describe the maximum size of the overall buffer. We are only binding 16 bytes from the buffer to the block here.

File: sdk/tests/conformance2/uniforms/large-uniform-buffers.html (hunk 1 above)
Comment (kenrussell) [reply sdk/tests/conformance2/uniforms/large-uniform-buffers.html]: At least could you please add an assertion that offs1 is evenly divisible by UNIFORM_BUFFER_OFFSET_ALIGNMENT? Thanks.

File: sdk/tests/conformance2/uniforms/large-uniform-buffers.html (hunk 2 above)
Comment (kenrussell) [reply sdk/tests/conformance2/uniforms/large-uniform-buffers.html]: Thanks for confirming. That's my understanding too.

**Comment by jdashg** [comment]
Don't worry too much about it. I figure I'll circle around down the road and fix-up some stragglers.

**Comment by kenrussell** [comment]
OK, thanks @jdashg .

File: sdk/tests/conformance2/uniforms/large-uniform-buffers.html (hunk 1 above)
Comment (JasperNV) [reply sdk/tests/conformance2/uniforms/large-uniform-buffers.html]: Sounds good. Any recommended ways to test an assertion? I'll tackle this tomorrow.

Comment (kenrussell) [reply sdk/tests/conformance2/uniforms/large-uniform-buffers.html]: const alignment = gl.getParameter(gl.UNIFORM_BUFFER_OFFSET_ALIGNMENT);
if (offs1 % alignment != 0) {
  testFailed("Platform has a strange uniform buffer offset alignment: " + alignment);

or similar.

**Comment by JasperNV** [comment]
Adjusted for feedback; apologies for the delay.

**Pull request merged**